nym-wallet/src-tauri/src/operations/signatures/sign.rs

```diff
@@ -0,0 +1,153 @@
+use k256::ecdsa::signature::Verifier;
+use serde::Serialize;
+use serde_json::json;
+
+#[derive(Debug, Serialize)]
+pub struct SignatureOutputJson {
+    pub account_id: String,
+    pub public_key: PublicKey,
+    pub signature: String,
+}
+
+#[tauri::command]
+pub async fn sign(
+    message: String,
+    state: tauri::State<'_, WalletState>,
+) -> Result<String, BackendError> {
+    let guard = state.read().await;
+    let client = guard.current_client()?;
+    let wallet = client.nymd.signer();
+    let derived_accounts = wallet.try_derive_accounts()?;
+    match derived_accounts.first() {
+        Some(account) => {
+            log::info!("<<< Signing message");
+            let signature = wallet.sign_raw_with_account(account, message.as_bytes())?;
+            let signature_as_hex_string = signature.to_string();
+            let output = SignatureOutputJson {
+                account_id: account.address().to_string(),
+                public_key: account.public_key(),
+                signature: signature_as_hex_string.to_string(),
+            };
+            log::info!(">>> Signing data {}", json!(output),);
+            Ok(signature_as_hex_string)
+        }
+        None => {
+            log::error!(">>> Unable to derive account");
+            Err(BackendError::SignatureError(
+                "unable to derive account".to_string(),
+            ))
+        }
+    }
+}
+
+async fn get_pubkey_from_account_address(
+    address: &AccountId,
+    state: &tauri::State<'_, WalletState>,
+) -> Result<PublicKey, BackendError> {
+    log::info!("Getting public key for address {} from chain...", address);
+    let guard = state.read().await;
+    let client = guard.current_client()?;
+    match client.nymd.get_account_details(address).await? {
+        Some(account) => {
+            let base_account = account.try_get_base_account()?;
+            if let Some(k) = base_account.pubkey {
+                log::info!("public key found {}", k.to_json());
+                Ok(k)
+            } else {
+                log::error!("No account associated with address {}", address);
+                Err(BackendError::SignatureError(format!(
+                    "No account associated with address {}",
+                    address
+                )))
+            }
+        }
+        None => {
+            log::error!("No account associated with address {}", address);
+            Err(BackendError::SignatureError(format!(
+                "No account associated with address {}",
+                address
+            )))
+        }
+    }
+}
+
+enum VerifyInputKind {
+    PublicKey(PublicKey),
+    AccountAddress(String),
+    CurrentAccountAddress,
+}
+
+impl TryFrom<Option<String>> for VerifyInputKind {
+    type Error = BackendError;
+
+    fn try_from(value: Option<String>) -> Result<Self, Self::Error> {
+        if value.is_none() {
+            return Ok(VerifyInputKind::CurrentAccountAddress);
+        }
+        let key = value.unwrap();
```

Comment:
Suggestion: avoid the `unwrap` with either match or `if let Some`

Reply:
Yes, thanks, this is what I usually do (or using alternative like `unwrap_or_else`), but in this specif case, because there is a test for `None` just above that returns, `unwrap` here will never panic.
And this avoids wrapping the whole next code block inside a match or `if let Some`.